### PR TITLE
fix ignored typos due to 'allowCompoundWords:true' setting

### PR DESCRIPTION
**Pull request by OnkarRuikar**
- addresses https://github.com/mdn/content/pull/35954#issuecomment-2365015752

The `allowCompundWords` setting in cSpell allows words like `writingsuggestions` and `popovertargetaction`. Without the setting, we would have to add these words to one of the ignore lists. But this has caused quite a lot of typos to fly under the radar, such as `progess`, `demonstates`, `diragram` etc.

The PR fixes missed typos.

***

~Also, with `allowCompundWords` on, we'll have to mark these words to be flagged explicitly. This has been done in the [second commit](https://github.com/mdn/content/commit/c6f6d0ddd6533b7232ff2abc89bb9363b8fdca60), but we'll have to keep looking for such words in the future and adding them to the lists.~

I suggest, instead of doing the above, we turn off the setting and explicitly add found compound words to terms/ignore lists. And camelCase the variable names as much as possible. We'll have to add the following existing 241 compound words to the ignore lists:
```plain
aaabcc
aabaac
aacbbbcac
ACCEPTEDTOS
accname
adduser
adjtimescapheight
adjtimesexheight
adrianroselli
advertisementreceived
aggregatable
alastairc
alistapart
ALLOWFROM
alphanumerals
Animometer
Annonay
antialiasing
apacheconf
apachectl
ASCIIID
assetlinks
audiocontext
beforeinstallpromptevent
beforetoggle
belgin
blockable
Blockable
brians
Browsersync
browsingtopics
bytestring
CACHEDIR
catchable
Caterina
characterboundsupdate
Cheatsheet
checkboard
checkmarks
clipboardchange
Codelab
collectability
collinearity
concatenator
contenido
contentlost
contentscripts
contentvisibilityautostatechange
controlbar
cookieless
cooluser
crashextensions
credentialled
crookedtimber
cssgridgarden
csstricks
Cubehelix
cubemaps
currentscreenchange
customstateset
cycletracker
davidwalsh
demuxing
Demuxing
Descripton
dropshow
enablei
Engish
Ethere
everytime
evilsite
examplestyles
Fairchild
fencedframe
fencedframeconfig
fencedframes
filetext
filterfunctions
Flagfox
forwidth
gattserverdisconnected
generatable
GETMATCHEDRULES
givenname
greenblue
greenzero
growshrink
harpercollins
highspeed
imagebad
imagegrid
infobox
infoboxes
infobox's
innertext
interarrival
ipados
jamiedoe
Jinping
Kobabe
lazyloaded
letmein
lightpurple
Lindenberg
linecaps
Linecaps
longwinded
loopstarted
lowdelay
macaron
Malala
mathit
maxplaybackrate
megastore
metastring
miplevel
Miroslav
Misparsed
modulepreloaded
montecarlo
mouseentry
multifactor
multimemory
mygroupmarker
mysampler
mystatename
navigables
nextslide
Nodepad
noteblocks
notrendered
Nowrapping
offsetted
ontextformatupdate
ontextupdate
pagereveal
pageswap
paintable
paintsize
papermakers
permissionspolicy
pipline
Platformer
playbackrate
popovertarget
popovertargetaction
precache
precached
precaches
precaching
Precaching
preconnects
preloaders
premultiplication
prerenderingchange
prerenders
previousslide
pricelist
properities
pythonanywhere
pythonaware
rawupdate
readonlyinline
realpython
redblue
redyellow
redzero
requireinteraction
reselecting
Resizeable
resnapped
Ronaldo
rtctransform
samplepay
scanability
screenreaders
screenschange
seecompattable
selectchange
selectedness
Serpentina
shadowrootdelegatesfocus
shadowrootmode
shadowrootserializable
sharedmem
sharpyuv
shortlinks
Shortlinks
shrinkwap
shrinkwrapping
sinkchange
smashingmagazine
Snapshotting
somestate
srihash
stackexchange
Stormys
Stratford
styleable
subformula
subformulas
subheaders
subpattern
subpopover
subproject
Subsentence
subsetting
subteam
subteams
sunsetting
supersets
swashoff
targettable
textformatupdate
textupdate
timeframe
truthyness
tunetheweb
tutsplus
underliner
underlyingly
usecase
useinbandfec
valuechange
viewbox
virtualkeyboardpolicy
webappmanifest
webbundle
webcodecs
webdav
webdesignerdepot
webidentity
webtransport
whitepoint
wisedog
workgroups
Workgroups
writingsuggestions
zeroblue
zeroyellow
zerozero
```

### CODE REVIEW

**Comment by Josh-Cena** [comment]
I don't think having a "blocklist" of typos is a good idea. I think we should turn off `allowCompoundWords` and add those existing compound words to the dictionary. What do you think?

File: files/en-us/web/mathml/element/mfenced/index.md [hunk 1]
```diff
@@ -10,7 +10,7 @@ browser-compat: mathml.elements.mfenced
 
 {{MathMLRef}}{{Deprecated_Header}}{{Non-standard_Header}}
 
-The **`<mfenced>`** [MathML](/en-US/docs/Web/MathML) element provides the possibility to add custom opening and closing brackets (such as parenthese) and separators (such as commas or semicolons) to an expression.
+The **`<mfenced>`** [MathML](/en-US/docs/Web/MathML) element provides the possibility to add custom opening and closing brackets (such as parenthesis) and separators (such as commas or semicolons) to an expression.
```
Comment (fred-wang) [inline files/en-us/web/mathml/element/mfenced/index.md:13]: I guess "parentheses"?

Comment (OnkarRuikar) [reply files/en-us/web/mathml/element/mfenced/index.md:13]:
```suggestion
The **`<mfenced>`** [MathML](/en-US/docs/Web/MathML) element provides the possibility to add custom opening and closing brackets (such as parentheses) and separators (such as commas or semicolons) to an expression.
```

File: .vscode/ignore-list.txt [hunk 2]
```diff
@@ -1,3 +1,244 @@
...
+cssgridgarden
+csstricks
+Cubehelix
+cubemaps
+currentscreenchange
+customstateset
+cycletracker
+davidwalsh
+demuxing
+Demuxing
```
Comment (bsmth) [inline .vscode/ignore-list.txt:64]: I don't believe we need this twice when there's two variants: first letter uppercase / lowercase, isn't that right?

File: .vscode/ignore-list.txt [hunk 3]
```diff
@@ -1,3 +1,244 @@
...
+apacheconf
+apachectl
+ASCIIID
+assetlinks
+audiocontext
+beforeinstallpromptevent
+beforetoggle
+belgin
+blockable
+Blockable
```
Comment (bsmth) [inline .vscode/ignore-list.txt:28]: Same comment as below

File: .vscode/ignore-list.txt [hunk 4]
```diff
@@ -1,3 +1,244 @@
...
+selectchange
+selectedness
+Serpentina
+shadowrootdelegatesfocus
+shadowrootmode
+shadowrootserializable
+sharedmem
+sharpyuv
+shortlinks
+Shortlinks
```
Comment (bsmth) [inline .vscode/ignore-list.txt:188]: Same as above

File: .vscode/ignore-list.txt (hunk 3 above)
Comment (OnkarRuikar) [reply .vscode/ignore-list.txt:28]:
```suggestion
```

File: files/en-us/web/api/svglengthlist/index.md [hunk 5]
```diff
@@ -41,7 +41,7 @@ An `SVGLengthList` object is indexable and can be accessed like an array.
 
 ### Using SVGLengthList
 
-An `SVGLengthList` object can be retrived from an {{domxref("SVGAnimatedLengthList")}} object, which itself is retrievable from many animatable length attributes, such as {{domxref("SVGTextPositioningElement.x")}}.
+An `SVGLengthList` object can be retrievd from an {{domxref("SVGAnimatedLengthList")}} object, which itself is retrievable from many animatable length attributes, such as {{domxref("SVGTextPositioningElement.x")}}.
```
Comment (bsmth) [inline files/en-us/web/api/svglengthlist/index.md:44]:
```suggestion
An `SVGLengthList` object can be retrieved from an {{domxref("SVGAnimatedLengthList")}} object, which itself is retrievable from many animatable length attributes, such as {{domxref("SVGTextPositioningElement.x")}}.
```

File: .vscode/ignore-list.txt (hunk 2 above)
Comment (OnkarRuikar) [reply .vscode/ignore-list.txt:64]:
```suggestion
```

File: .vscode/ignore-list.txt (hunk 4 above)
Comment (OnkarRuikar) [reply .vscode/ignore-list.txt:188]:
```suggestion
```

File: .vscode/ignore-list.txt [hunk 6]
```diff
@@ -1,3 +1,244 @@
...
+webbundle
+webcodecs
+webdav
+webdesignerdepot
+webidentity
+webtransport
+whitepoint
+wisedog
+workgroups
+Workgroups
```
Comment (OnkarRuikar) [inline .vscode/ignore-list.txt:237]:
```suggestion
```

File: .vscode/ignore-list.txt [hunk 7]
```diff
@@ -1,3 +1,244 @@
...
+pipline
+Platformer
+playbackrate
+popovertarget
+popovertargetaction
+precache
+precached
+precaches
+precaching
+Precaching
```
Comment (OnkarRuikar) [inline .vscode/ignore-list.txt:151]:
```suggestion
```

File: files/en-us/web/css/css_shapes/overview_of_shapes/index.md [hunk 8]
```diff
@@ -96,7 +96,7 @@ In the example below we have added a `shape-margin` to a basic shape. Change the
 
 In the examples above, we have used images or a visible element to define the shape, meaning that you can see the shape on the page. Instead, you might want to flow some text along a non-rectangular invisible line. We could, for example, add an empty floated {{htmlelement("div")}} or {{htmlelement("span")}} element to our DOM and make it invisible. However, we can create a shape with only CSS using [generated content](/en-US/docs/Learn/CSS/Howto/Generated_content) and keep all our styling functionality inside the CSS.
 
-In this example, we use generated content to insert an element with a height and width of 150px. We can then use basic Sshapes, box values, or even the alpha channel of an image to create a shape for the text to wrap around.
+In this example, we use generated content to insert an element with a height and width of 150px. We can then use basic Shapes, box values, or even the alpha channel of an image to create a shape for the text to wrap around.
```
Comment (bsmth) [inline files/en-us/web/css/css_shapes/overview_of_shapes/index.md:99]: I think the capital 'S' is the typo:

```suggestion
In this example, we use generated content to insert an element with a height and width of 150px. We can then use basic shapes, box values, or even the alpha channel of an image to create a shape for the text to wrap around.
```

File: files/en-us/web/javascript/reference/errors/unexpected_token/index.md [hunk 9]
```diff
@@ -63,7 +63,7 @@ Sometimes, you leave out parentheses around `if` statements:
 
 ```js-nolint example-bad
 function round(n, upperBound, lowerBound) {
-  if (n > upperBound) || (n < lowerBound) { // Not enough parenthese here!
+  if (n > upperBound) || (n < lowerBound) { // Not enough parenthesis here!
```
Comment (bsmth) [inline files/en-us/web/javascript/reference/errors/unexpected_token/index.md:66]:
```suggestion
  if (n > upperBound) || (n < lowerBound) { // Missing parentheses here!
```

Comment (OnkarRuikar) [reply files/en-us/web/javascript/reference/errors/unexpected_token/index.md:66]: https://www.merriam-webster.com/dictionary/parenthesis
```suggestion
  if (n > upperBound) || (n < lowerBound) { // Missing a parenthesis here!
```

**Review by bsmth — APPROVED** [review]
LGTM, thank you 👍🏻

File: files/en-us/web/javascript/reference/errors/unexpected_token/index.md [hunk 10]
```diff
@@ -63,7 +63,7 @@ Sometimes, you leave out parentheses around `if` statements:
 
 ```js-nolint example-bad
 function round(n, upperBound, lowerBound) {
-  if (n > upperBound) || (n < lowerBound) { // Not enough parenthese here!
+  if (n > upperBound) || (n < lowerBound) { // Missing a parenthesis here!
```
Comment (OnkarRuikar) [inline files/en-us/web/javascript/reference/errors/unexpected_token/index.md:66]:
```suggestion
if (n > upperBound) || (n < lowerBound) { // Missing parentheses here!
```

File: files/en-us/web/javascript/reference/errors/unexpected_token/index.md (hunk 9 above)
Comment (bsmth) [reply files/en-us/web/javascript/reference/errors/unexpected_token/index.md]: It's missing a pair, but I see the dictionary definition says "one or both"

Comment (OnkarRuikar) [reply files/en-us/web/javascript/reference/errors/unexpected_token/index.md]: It's a bit confusing: `parenthesis` is singular, and `parentheses` is plural.

File: files/en-us/web/javascript/reference/errors/not_a_function/index.md [hunk 11]
```diff
@@ -126,7 +126,7 @@ const myNewDog = new Dog();
 myNewDog.name("Cassidy"); //Dog { age: 11, color: 'black', dogName: 'Cassidy' }
 ```
 
-### Using parenthese for multiplication
+### Using parenthesis for multiplication
```
Comment (Josh-Cena) [inline files/en-us/web/javascript/reference/errors/not_a_function/index.md:129]:
```suggestion
### Using parentheses for multiplication
```

**Review by Josh-Cena — APPROVED** [review]
This is awesome! I'm excided about all the new typos.

**Pull request merged**